src/splunk_otel/profile.py

```diff
@@ -127,8 +132,10 @@ def wrapper(wrapped, _instance, args, kwargs):
 def _collect_stacktraces():
     out = []
     frames = sys._current_frames()  # noqa SLF001
-
+    profile_scraper_thread_id = threading.get_ident()
     for thread_id, frame in frames.items():
+        if thread_id == profile_scraper_thread_id:
```

Comment:
logic changed but test didn't?

Reply:
Yeah, there aren't tests that exercise extracting the current stack frame (seemed problematic). I'll give that another look for a separate PR.